### PR TITLE
Support batch commit and persisting encrypted votes

### PR DESCRIPTION
**Pull request by rawfalafel**
~STILL A WORK IN PROGRESS~
- [x] Add tests for `batchCommit`

### CODE REVIEW

**Comment by rawfalafel** [comment]
FWIW, the current compiled contract has a bytecode size of 24991 bytes. The bytecode size limit seems to be ~26469 bytes, so we're getting pretty close.

The lowest hanging fruit to shave off bytes would be to convert error strings into error codes (e.g. `IN_COMMIT_PHASE` vs `"Cannot reveal while in the commit phase"`)

**Comment by mrice32** [comment]
I also agree wrt error codes - we can cross that bridge when we come to it, though. We also have a few libraries that we can make external, although I'm not sure how much space that'll save.

**Comment by ptare** [comment]
[Just an idea] What do you guys think about storing `encryptedVote` in `VoteInstance` directly (keyed by the voter's address), getting rid of `EncryptedSender`?

We'd lose the ability to "send" messages to another address, but in this context, I can't think of great uses for sending a message that you cannot read back yourself, suggesting that the reader and sender have to always be the same.

**Comment by mrice32** [comment]
What would the advantage of that approach? Smaller bytecode footprint?

**Comment by ptare** [comment]
> What would the advantage of that approach? Smaller bytecode footprint?

Eliminating features that we don't use, such as authorized senders and the topics.

**Comment by rawfalafel** [comment]
> Eliminating features that we don't use, such as authorized senders and the topics.

Yeah, it could be argued that `EncryptedSender` is too general for our use case. That having been said, it's kind of nice to separate this logic into a separate contract in case the requirements change. I lean towards keeping the `EncryptedSender` for now with the goal of stripping it down to only meet our requirements before the testnet release.

**Comment by mrice32** [comment]
I think the separation of the logic is nice, especially since the purpose is pretty distinct from the core voting logic. I don't think the current implementation is overly complex, so I think it's safe just to leave as is until we get more user feedback on the AVS and dApp since we're pretty sure that the AVS as it stands isn't super user-friendly.

**Comment by rawfalafel** [comment]
This PR's ready. PTAL.

File: core/test/Voting.js [hunk 1]
```diff
@@ -1012,4 +1014,141 @@ contract("Voting", function(accounts) {
...
+    await moveToNextPhase(voting);
+
+    const decryptedMessage = await decryptMessage(privateKey, retrievedEncryptedMessage);
+    const retrievedVote = JSON.parse(decryptedMessage);
+
+    assert(await didContractThrow(voting.revealVote(identifier, time, getRandomSignedInt(), getRandomUnsignedInt())));
+    await voting.revealVote(identifier, time, retrievedVote.price, retrievedVote.salt);
+  });
+
+  it("Commit and persist multiple messages", async function() {
```
Comment (mrice32) [inline core/test/Voting.js]: nit: can we change the title to note that this test rewrites a single commit rather than commits twice?

**Pull request merged**